src/app/components/accountHeader/index.tsx

```diff
@@ -154,6 +154,11 @@ function AccountHeaderComponent({
     setShowOptionsDialog(false);
   };
 
+  const handleLockWallet = async () => {
+    await lockWallet();
+    navigate('/login');
```

Comment:
Won't this be handled by the guard?

Reply:
since the seed phrase and encrypted seed are no longer considered global state the `Authguard` won't re-render and trigger the effect when locking the vault the only alternative here being adding a listener on storage and storing a flag on the wallet store

Reply:
Oh, that's not ideal. We want to be reactive to changes in the seed vault. I think we should make the following changes:
- In the seed vault repo in core, we add an event publisher to the seedvault class that you can listen on and whenever a function that causes a change in the seedvault is fired, we notify the listeners.
- In the extension/mobile app, we create a singleton of the seedvault and expose it via Context API. We can then change the useSeedVault hook to wrap useContext and we provide the singleton seedvault as the value in the provider.
- The context then listens to changes to the seedvault and regenerates the singleton to a new instance of the seedvault and passes it down as the new value. retriggering anything that's using the seedvault hook.

What do you think?